shared/domains/chelonia/chelonia.js

```diff
@@ -290,25 +386,25 @@ sbp('sbp/selectors/register', {
     }
   },
   'chelonia/latestContractState': async function (contractID: string) {
-    const events = await sbp('chelonia/out/eventsSince', contractID, contractID)
-    let state = {}
+    const events = await sbp('chelonia/private/out/eventsSince', contractID, contractID)
+    let state = cloneDeep(sbp(this.config.stateSelector)[contractID] || Object.create(null))
```

Comment:
Oh, I missed your reply:

> It's incorrect to start with an empty state if there already is a state. The original code caused the _volatile state to be missing.

OK, the intent of this function is to retrieve the state of contracts that you're not subscribed to...

If you are already subscribed to a contract, then you can simply return its state.

So in this case you wouldn't clone the state, and you certainly wouldn't re-apply all of the events on top of that state, and that would corrupt it.

Reply:
Thanks, I'll look into this further. It was causing issues before when starting out empty, but I don't think I tried returning the existing state.

Reply:
Yeah, and thinking on it, if the state exists because it's a subscribed contract, please go ahead and return a cloned copy of it, as that is more in line with how `latestContractState` is expected to work (returning a state snapshot of sorts).

Reply:
This was never addressed...